### ISSUE TITLE
Concurrent calls

### ISSUE THREAD

**Issue by jurafa** [jprjr/lua-resty-exec#8]
I use lua-resty-exec with openresty on Raspberry Pi.

I sometimes observe strange behavior when multiple requests are supposed to be handled concurrently.
I, therefore, started with testing the framework with ApacheBench.
The lua script is very simple:

    -- prog.argv = {'ip', 'addr'}
    prog.argv = {'sleep', '0.01'}
    local res, err = prog()
    if err ~= nil then
        ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
    else
        ngx.exit(ngx.HTTP_OK)
    end

This is an example of calling the ApacheBench:
`ab -k -c 10 -n 1000 -s 30 127.0.0.1:8010/test`

When this test is executed without concurrency (-c 1) everything works fine.
However, if I run the test with higher concurrency (even -c 2 is usually sufficient) the framework starts to timeout. Usually, the *ip addr* causes more troubles.

The *sockexec* program prints
`Crap: can't find conn_id for pid xxxx`
to the terminal.

Any help or advice is appreciated.

**Comment by jprjr** [comment]
Hey there - I just put out sockexec 1.2.0, which should fix this concurrency problem.

Basically, I had logic like "did the child process exit? Close the connection immediately", so sometimes it wouldn't grab stdout/stderr because I process signals before processing child processes/connections/etc. Now I wait until the child's stdout/stderr has closed before terminating the connection.

Another error - there's a loop for dealing with child processes that have exited and figuring out which connection they're a part of, I wasn't resetting the loop variable correctly. That caused the "Crap: can't find conn_id for pid xxxx" errors.

I also went through and cleaned up the error reporting to have more consistent output, and I added a '-d' (debug) switch to increase the logging verbosity.

Lastly, there was a hard-coded limit of 10 queued connections, I bumped that default to 100, and I made it configurable at runtime (via the -q flag).

Give it a try and let me know if it works!
